eclipse-theia/theia
# testing: Add Theia Playwright framework (#10337)

Fixes https://github.com/eclipse-theia/theia/issues/10337



#### What it does


This change adds a page object framework based on Playwright that enables writing system tests (or end-to-end tests) of Theia-based applications more efficiently by introducing an abstraction layer over Theia's user interfaces, encapsulating the details of the user interface interactions, wait conditions, etc. This helps keeping your tests more concise, maintainable, and stable.

As part of this change, we also add tests for the introduced page objects, which run against the Theia example browser app in this repo. Currently, the framework is added as a separate package, which in a next step should be published and made consumable via npm, which we are happy to work on once this is merged.

#### How to test


With this change we added some initial documentation in the form of MD files on how to build, run and test the page object framework. Basically you'll have to run the following commands in the theia-playwright-testframework folder:

    yarn
    yarn theia:start &
    yarn ui-tests

There are other scripts for running the tests headfully or only run specific tests (see documentation).

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
